Kinetix.SpaServiceGenerator/Program.cs

```diff
@@ -61,9 +74,18 @@ internal class Program
                 var syntaxTree = await controller.GetSyntaxTreeAsync();
                 var controllerClass = GetClassDeclaration(syntaxTree);
 
+                IReadOnlyList<string> folders = null;
+#if NET471
+                folders = controller.Folders;
+#endif
+#if NETCOREAPP2_1
+                var parts = Path.GetRelativePath(controller.Project.FilePath, controller.FilePath).Split(@"\");
+                folders = parts.Skip(1).Take(parts.Length - 2).ToList();
+#endif
+
                 var firstFolder = frontEnds.Count() > 1 ? $"/{controller.Project.Name.Split('.')[1].ToDashCase()}" : string.Empty;
-                var secondFolder = controller.Folders.Count > 1 ? $"/{string.Join("/", controller.Folders.Skip(1).Select(f => f.ToDashCase()))}" : string.Empty;
-                var folderCount = (frontEnds.Count() > 1 ? 1 : 0) + controller.Folders.Count;
+                var secondFolder = folders.Count > 1 ? $"/{string.Join("/", folders.Skip(1).Select(f => f.ToDashCase()))}" : string.Empty;
+                var folderCount = (frontEnds.Count() > 1 ? 1 : 0) + folders.Count - 1;

```

Comment:
This is used to generate the JS imports in the service files, and I noticed a "-1" was missing (my `Folders.Count` was always 0 instead of always 1, so I didn't catch it).
I hope this works for your folder hierarchy